src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -175,19 +178,52 @@ protected StarlarkBaseExternalContext(
     this.remoteExecutor = remoteExecutor;
     this.asyncTasks = new ArrayList<>();
     this.allowWatchingPathsOutsideWorkspace = allowWatchingPathsOutsideWorkspace;
+    this.executorService =
+        Executors.newThreadPerTaskExecutor(
+            Thread.ofVirtual()
+                .name("downloads-" + workingDirectory.getBaseName())
```

Comment:
```suggestion
                .name("downloads-" + getIdentifyingStringForLogging() + "-", 0)
```

Reply:
alternatively, `"downloads[" + getIdentifyingStringForLogging() + "]-"`, if using just dashes makes visual parsing difficult.

Reply:
This doesn't quite work as `getIdentifyingStringForLogging` throws an NPE when called in the constructor. Should I pass it down as a constructor argument or use `workingDirectory.getBaseName()` instead?

Reply:
oh right. let's just make it a `protected final` field, IMO